File: amazia-console/app/Product/Controller/DeleteProductController.php
<?php

namespace App\Product\Controller;

use App\Http\Controllers\Controller;
use App\Product\Service\DeleteProductService;

class DeleteProductController extends Controller
{
    private DeleteProductService $service;

    public function __construct(DeleteProductService $service)
    {
        $this->service = $service;
    }

    public function __invoke(int $id)
    {
        $response = $this->service->delete($id);

        if ($response->status() === 204) {
            return response()->noContent();
        }

        return response()->json($response->json(), $response->status());
    }
}

File: amazia-core/src/main/java/com/example/product/controller/DeleteProductController.java
package com.example.product.controller;

import com.example.product.service.DeleteProductService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api")
public class DeleteProductController {

    private final DeleteProductService deleteProductService;

    public DeleteProductController(DeleteProductService deleteProductService) {
        this.deleteProductService = deleteProductService;
    }

    @DeleteMapping("/products/{id}")
    public ResponseEntity<Void> delete(@PathVariable Long id) {
        deleteProductService.delete(id);
        return ResponseEntity.noContent().build();
    }
}
